src/components/leaf.js

```diff
@@ -88,23 +88,16 @@ class Leaf extends React.Component {
     const text = this.renderText(props)
     if (el.textContent != text) return true
 
-    // If the selection was previously focused, and now it isn't, re-render so
-    // that the selection will be properly removed.
-    if (this.props.state.isFocused && props.state.isBlurred) {
-      const { index, node, ranges, state } = this.props
-      const { start, end } = OffsetKey.findBounds(index, ranges)
-      if (state.selection.hasEdgeBetween(node, start, end)) return true
-    }
-
-    // If the selection will be focused, only re-render if this leaf contains
-    // one or both of the selection's edges.
-    if (props.state.isFocused) {
+    // If the selection will be focused and native selection is not already
+    // in syncro with it, update native selection.
+    if (props.state.isFocused && props.selectionNeedsRedraw) {
       const { index, node, ranges, state } = props
       const { start, end } = OffsetKey.findBounds(index, ranges)
-      if (state.selection.hasEdgeBetween(node, start, end)) return true
+      if (state.selection.hasEdgeBetween(node, start, end)) {
+        this.updateSelection(props)
```

Comment:
I agree with you, it's a bit weird, but we can't really do otherwise at moment.
To avoid this we should rise the selection update code up in the component hierarchy and this would make the `Node` component source a mess unless we move the `Text` node related content in a separated component.